src/hubextensions.ts

```diff
@@ -30,26 +31,74 @@ export function __PRIVATE__wrapStartTransactionWithProfiling(startTransaction: S
     const profile_id = uuid4();
     const uniqueTransactionName = `${transactionContext.name} ${profile_id}`;
 
-    // profilesSampleRate is multiplied with tracesSampleRate to get the final sampling rate.
+    // profilesSampleRate is multiplied with tracesSampleRate to get the final sampling rate. We dont perform
+    // the actual multiplication to get the final rate, but we discard the profile if the transaction was sampled,
+    // so anything after this block from here is based on the transaction sampling.
     if (!transaction.sampled) {
       return transaction;
     }
 
-    // @ts-expect-error profilesSampleRate is not part of the options type yet
-    const profilesSampleRate = this.getClient()?.getOptions?.()?.profilesSampleRate;
-    if (profilesSampleRate === undefined) {
+    const client = this.getClient();
+    if (!client) {
+      if (isDebugBuild()) {
+        logger.log('[Profiling] Profiling disabled, no client found.');
+      }
+      return transaction;
+    }
+    const options = client.getOptions();
+    if (!options) {
+      if (isDebugBuild()) {
+        logger.log('[Profiling] Profiling disabled, no options found.');
+      }
+      return transaction;
+    }
+
+    // @ts-expect-error sampler is not yer part of the sdk options
+    const profilesSampler = options.profilesSampler;
+    // @ts-expect-error sampler is not yer part of the sdk options
+    let profilesSampleRate: number | undefined = options.profilesSampleRate;
+
+    // Prefer sampler to sample rate if both are provided.
+    if (typeof profilesSampler === 'function') {
+      profilesSampleRate = profilesSampler(customSamplingContext);
+    } else if (customSamplingContext && customSamplingContext['parentSampled'] !== undefined) {
+      profilesSampleRate = customSamplingContext['parentSampled'];
```

Comment:
Is this the sampling decision on the parent transaction? If that is the case, you're effectively saying sample all transactions which I don't think is the intention here.

Reply:
Yes correct, I took this logic off tracing, but I could not find the actual reasoning for inheriting this from https://github.com/getsentry/sentry-javascript/pull/3068

Reply:
So if I'm correct, then the reason for this is because for distributed tracing, when the initial transaction is sampled, we would want the entire trace to be sampled. i.e. the browser page load transaction is sampled, so we want to propagate that sampling decision to all the API calls so that those transactions are also sampled.

I don't think this is what we want to do in profiling, at least not right now. Because it's effectively ignoring the profile samples rate and saying profile 100% of transactions. Now, this may be something we want but it's definitely not something we've discussed since we've been focused on profiling individual platforms not if we want to profile the entire trace.

Reply:
Ahhhhh, ok. I was not sure what the use case was, thanks for explaining! I will remove it for now, but like you mentioned, this may be something we want to actually support and is worth bringing up to the team.

Reply:
FWIW, if we are to implement this inheriting, we should not inherit the trace's sampling decision. We would want to do something like pass a profiling sampling decision from the root transaction alongside the transaction's sampling decision so that the whole trace is profile but only when the root transaction is profiled.

But again, this is something we should discuss before moving forwards.

Reply:
@Zylphrex I just remove the code, no reason to keep it as it's easy to add back whenever.